openmc/stats/univariate.py

```diff
@@ -1312,40 +1339,56 @@ def integral(self):
         ])
 
     def clip(self, tolerance: float = 1e-6, inplace: bool = False) -> Mixture:
-        r"""Remove low-importance points from contained discrete distributions.
+        r"""Remove low-importance points / distributions
 
-        Given a probability mass function :math:`p(x)` with :math:`\{x_1, x_2,
-        x_3, \dots\}` the possible values of the random variable with
-        corresponding probabilities :math:`\{p_1, p_2, p_3, \dots\}`, this
-        function will remove any low-importance points such that :math:`\sum_i
-        x_i p_i` is preserved to within some threshold.
+        Like :meth:`Discrete.clip`, this method will remove low-importance
+        points from discrete distributions contained within the mixture but it
+        will also clip any distributions that have negligible contributions to
+        the overall intensity.
 
         .. versionadded:: 0.14.0
 
         Parameters
         ----------
         tolerance : float
-            Maximum fraction of :math:`\sum_i x_i p_i` that will be discarded
-            for any discrete distributions within the mixture distribution.
+            Maximum fraction of intensities that will be discarded.
         inplace : bool
             Whether to modify the current object in-place or return a new one.
 
         Returns
         -------
-        Discrete distribution with low-importance points removed
+        Distribution with low-importance points / distributions removed
 
         """
+        # Determine indices for any distributions that contribute non-negligibly
+        # to overall intensity
+        intensities = [prob*dist.integral() for prob, dist in
+                       zip(self.probability, self.distribution)]
+        indices = _intensity_clip(intensities, tolerance=tolerance)
+
         if inplace:
+            # Clip mixture of distributions
+            self.probability = self.probability[indices]
+            self.distribution = [self.distribution[i] for i in indices]
+
+            # Clip points from Discrete distributions
             for dist in self.distribution:
                 if isinstance(dist, Discrete):
                     dist.clip(tolerance, inplace=True)
+
             return self
         else:
+            # Clip mixture of distributions
+            probability = self.probability[indices]
+            distribution = [self.distribution[i] for i in indices]
+
+            # Clip points from Discrete distributions
             distribution = [
                 dist.clip(tolerance) if isinstance(dist, Discrete) else dist
-                for dist in self.distribution
+                for dist in distribution
             ]
```

Comment:
This logic can come before the `if inplace... else` block if you add `inplace=inplace` to the `clip` call to reduce a little bit of duplicate code. Then we can check the new intensity and issue a warning if necessary. Then use the `if inplace... else` block to simply return the respective objects (with setting `self.probability = probability` and `self.distribution = distribution` in the `if` clause prior to `return self`.